src/service/invoice-service.ts

```diff
@@ -428,23 +356,25 @@ export default class InvoiceService {
     return this.manager.findOne(Invoice, options);
   }
 
-  async getSubTransactionsInvoice(invoice: Invoice, transactions: BaseTransactionResponse[], isCreditInvoice: boolean) {
-    const ids = transactions.map((t) => t.id);
-
-    let where: FindOptionsWhere<SubTransaction> = {
-      transaction: { id: In(ids) },
-    };
-    // If we have a credit invoice we filter out all unrelated subTransactions.
-    if (isCreditInvoice) where.to = { id: invoice.toId };
-    return  this.manager.find(SubTransaction, { where, relations: ['transaction', 'to', 'subTransactionRows',
-      'subTransactionRows.invoice', 'subTransactionRows.product', 'subTransactionRows.product.product', 'subTransactionRows.product.vat'] });
+  /**
+   * Removes the invoice reference from all sub transaction rows
+   * @param invoice
+   */
+  async removeSubTransactionInvoice(invoice: Invoice) {
+    const subTransactionRows = await this.manager.find(SubTransactionRow, { where: { invoice: { id: invoice.id } } });
+    const promises: Promise<any>[] = [];
+    subTransactionRows.forEach((tSubRow) => {
+      const row = tSubRow;
+      row.invoice = null;
+      promises.push(this.manager.save(SubTransactionRow, row));
+    });
+    return Promise.all(promises);
```

Comment:
Can't you just replace this complete function with an `onDelete: 'SET NULL'` on the `SubTransactionRow` side? Then, this will all be done automatically by the database (and you never make a mistake).